### PR TITLE
APICenter Extension Initial Release

### PR DESCRIPTION
**Pull request by arpishahmsft**
Creating a new extension for APICenter control plane apis for operations

1. Service - list, show, delete
2. workspace - create, update, list, show, delete
3. api - create, update, list, show, delete
4. api version - create, update, list, show, delete
5. api definition - create, update, list, show, delete, import and export specification (Json files)
6. deployment - create, update, list, show, delete
7. environment - create, update, list, show, delete
8. metadata schema - create, update, list, show, delete, export metadata schema (json file)
9. quick add or register operation - custom command
10. Custom command extensions for import/export specs, create/update metadata schema, export metadata schema, quick add/api register operations

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/apic-extension/azext_apic_extension/custom.py [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+
+import os
+import sys
+import json
+import requests
+from knack.log import get_logger
+import chardet
+from azure.cli.core.aaz import *
```
Comment (AllyW) [inline src/apic-extension/azext_apic_extension/custom.py:17]: `import` should be specific to detailed classes and funcs

File: src/apic-extension/azext_apic_extension/aaz/latest/apic/workspace/_update.py [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+                max_length=90,
+                min_length=1,
+            ),
+        )
+        _args_schema.workspace_name = AAZStrArg(
+            options=["-w", "--name", "--workspace", "--workspace-name"],
+            help="The name of the workspace.",
+            required=True,
+            id_part="child_name_1",
+            default="default",
```
Comment (AllyW) [inline src/apic-extension/azext_apic_extension/aaz/latest/apic/workspace/_update.py:67]: default value of update.py is not recommended

File: src/apic-extension/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Azure CLI ApicExtension Extension #
+This is an extension to Azure CLI to manage ApicExtension resources.
+
+## How to use ##
```
Comment (AllyW) [inline src/apic-extension/README.md:4]: please add cmd usage here.

Comment (arpishahmsft) [reply src/apic-extension/README.md]: Added. thank you!

File: src/apic-extension/azext_apic_extension/custom.py [hunk 4]
```diff
@@ -0,0 +1,480 @@
...
+            registered=True
+        )
+        return args_schema
+
+    def _output(self, *args, **kwargs):
+        result = self.deserialize_output(self.ctx.vars.instance, client_flatten=True)
+        arguments = self.ctx.args
+
+        if result:
+            print('Results found. Exporting to', arguments.source_profile)
```
Comment (AllyW) [inline src/apic-extension/azext_apic_extension/custom.py:92]: is `print` here useful? If needed, please use logger.debug or logger.warning

Comment (arpishahmsft) [reply src/apic-extension/azext_apic_extension/custom.py:92]: Yes, it is good for customer to know that results are found. That is why I used print. Thanks!

Comment (arpishahmsft) [reply src/apic-extension/azext_apic_extension/custom.py:92]: updated to logger.info

**Pull request merged**